README.md

```diff
@@ -438,6 +438,120 @@ const clientWithNoCache = await Client.create(wallet, {
+const creatorClient = await Client.create(yourSigner)
+creatorClient.enableGroupChat()
+```
+
+This enables the following capabilities required to provide group chat:
+
+- The client will be able to create group chats
+- Group chats will be present in `client.conversations.list()`
+- The client will understand GroupChat codecs such as `GroupChatMemberAdded` and `GroupChatTitleChanged`
+
+### Create a group chat
+
+Enable a user to create a group chat using `newGroupConversation` and adding member addresses to it:
+
+```ts
+const memberAddresses = [
+  '0x194c31cAe1418D5256E8c58e0d08Aee1046C6Ed0',
+  '0x937C0d4a6294cdfa575de17382c7076b579DC176',
+]
+const groupConversation =
+  creatorClient.conversations.newGroupConversation(memberAddresses)
+```
+
+Assuming the other members of the group chat have clients with group chat enabled, they'll see the new group chat in their conversation list.
+
+### Send a message to a group chat
+
+Enable a user to send a message to a group chat the same way you send messages to a 1:1 conversation:
+
+```ts
+await groupConversation.send('hello everyone')
+```
+
+### Load group chats
+
+When you enabled group chat for your Client, you enabled group chats to be returned in `conversations.list()`:
+
+```ts
+const conversations = await creatorClient.conversations.list()
+const conversation = conversations[0]
+
+console.log(conversation.isGroup) // => true when it's a group conversation
+```
+
+### Change the group chat title
+
+Enable a member of a group chat to change the group chat title by sending a message with the `GroupChatTitleChanged` content type:
+
+```ts
+import { ContentTypeGroupChatTitleChanged } from '@xmtp/xmtp-js'
+import type { GroupChatTitleChanged } from '@xmtp/xmtp-js'
+
+const
+```
+
+### Manage group state with the `GroupChat` class
+
+Use the `GroupChat` class to keep track of group state, such as the group title and member list:
+
+```ts
+const conversations = await creatorClient.conversations.list()
+const conversation = conversations[0] // assume this conversation is a group conversation
+
+const groupChat = new GroupChat(creatorClient, conversation)
+```
+
+You can also use the `GroupChat` class to change the group chat title and member list.
+
+#### Change a group chat title
+
+To change a group chat title, call `changeTitle` on a `GroupChat` instance:
+
+```ts
+await groupChat.changeTitle('The fun group')
+```
+
+This sends a message with the `GroupChatTitleChanged` content type to the group chat that clients can display. Clients can also use the message to update the group chat title on their end.
+
+#### Add a group chat member
+
+To add a group chat member, call `addMember` on a `GroupChat` instance:
+
+```ts
+await groupChat.addMember('0x194c31cAe1418D5256E8c58e0d08Aee1046C6Ed0')
+```
+
+This sends an invitation to the recipient address. It also send a `GroupChatMemberAdded` message that clients can display and use to update their gorup chat member lists.
+
+#### Rebuild the group state
```

Comment:
Just curious - when might a dev need to do this? After the group chat title or member list is updated?

Reply:
They'd need to do this the first time they loaded the group chat to make sure they have everything up to date.